### PR TITLE
ENG-1370 support for both old and new plugin descriptor format - QUIC…

### PR DESCRIPTION
**Pull request by firegloves**
…K FIX SOLUTION

I introduced also a fix to keep pod name length < 63 that is a k8s constraint

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **firegloves**

If they are not already assigned, you can assign the PR to them by writing `/assign @firegloves` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando-k8s%2Fentando-component-manager).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando-k8s/entando-component-manager/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by entando-jx** [comment]
@firegloves: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 99ec4e2ff74f907f61169e162370d5b14a151382 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@firegloves: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | a60d13e0abcdd2fb3e6d7dcab54dad8040800b24 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@firegloves: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 99ec4e2ff74f907f61169e162370d5b14a151382 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-component-manager-ecmpr68**

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-component-manager-ecmpr68**

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-component-manager-ecmpr68**

File: src/main/java/org/entando/kubernetes/model/bundle/descriptor/plugin/PluginDescriptorV1Spec.java
Comment (w-caffiero-entando) [inline src/main/java/org/entando/kubernetes/model/bundle/descriptor/plugin/PluginDescriptorV1Spec.java:11]: If the class is meant to be immutable consider using `@Value` instead of `@Data`

File: src/main/java/org/entando/kubernetes/service/digitalexchange/BundleUtilities.java
Comment (w-caffiero-entando) [reply src/main/java/org/entando/kubernetes/service/digitalexchange/BundleUtilities.java:81]: Wonder if a silent truncate is safe.
Is the plugin name supposed to be unique?

**Review by w-caffiero-entando — CHANGES_REQUESTED** [review]

File: src/main/java/org/entando/kubernetes/model/bundle/descriptor/plugin/PluginDescriptor.java
Comment (w-caffiero-entando) [reply src/main/java/org/entando/kubernetes/model/bundle/descriptor/plugin/PluginDescriptor.java:1]: I see, in order to keep this class and the new ones on the same package.

File: src/main/java/org/entando/kubernetes/model/bundle/descriptor/plugin/PluginDescriptorV1Spec.java
Comment (firegloves) [reply src/main/java/org/entando/kubernetes/model/bundle/descriptor/plugin/PluginDescriptorV1Spec.java:11]: It could be the rightest way but we are using jackson to parse the descriptors. Jackson should use reflection and therefore the empty constructor and then the setters. I think it's impossible to apply the immutable pattern using these libs

File: src/main/java/org/entando/kubernetes/service/digitalexchange/BundleUtilities.java
Comment (firegloves) [reply src/main/java/org/entando/kubernetes/service/digitalexchange/BundleUtilities.java:81]: This is an example of a pod name:
`lcorsettientando-the-lucas-0-0-1-server-deployment-58d859dpd5fp`

the segment `-server-deployment-58d859dpd5fp` is autogenerated by k8s and appended to the first part. length 30

Max accepted length is 63, so the segment created by us must have a max length = 33
This is the segment is created by us: `lcorsettientando-the-lucas-0-0-1` max length 33 (trimming the final hyphen to avoid a double -- that is not compatible)

Thanks to the k8s suffix we should be safe about name collisions, but we may lose the docker image version in the pod name, as you can see in the example above

**Comment by entando-jx** [comment]
@firegloves: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 402a50ba37a86243cf9751329d87b863e5573330 | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando-k8s/entando-component-manager/PR-68/9.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@firegloves: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | d3deb7215871e47fec325802aca51c62230333aa | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando-k8s/entando-component-manager/PR-68/10.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: src/main/java/org/entando/kubernetes/service/digitalexchange/BundleUtilities.java
Comment (w-caffiero-entando) [reply src/main/java/org/entando/kubernetes/service/digitalexchange/BundleUtilities.java]: > Max accepted length is 63, so the segment created by us must have a max length = 32

Please make it explicit on code.
Use constants and make explicit calculation.

> Thanks to the k8s suffix we should be safe about name collisions, but we may lose the docker image version in the pod name, as you can see in the example above

right, but still I'm not happy having something in the descriptor and something else installed.
I'd say that the installation should fail in this case.
We may try to reduce the possibility to trigger it by doing some on the CLIs

Comment (firegloves) [reply src/main/java/org/entando/kubernetes/service/digitalexchange/BundleUtilities.java]: maybe we can introduce a validation step that verifies the max length of the image field?

Comment (firegloves) [reply src/main/java/org/entando/kubernetes/service/digitalexchange/BundleUtilities.java]: and I may move the constants in the validator

Comment (w-caffiero-entando) [reply src/main/java/org/entando/kubernetes/service/digitalexchange/BundleUtilities.java]: @ampie @joewhite101 @ffleandro thougts?

Comment (w-caffiero-entando) [reply src/main/java/org/entando/kubernetes/service/digitalexchange/BundleUtilities.java]: @ampie do you think it's possible to set a max-len limit at CRD level on this field?

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-component-manager-ecmpr68**

**Comment by entando-jx** [comment]
@firegloves: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 8b5585d6576951181a5d78e78390960ce0acd93f | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando-k8s/entando-component-manager/PR-68/12.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-component-manager-ecmpr68**

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-component-manager-ecmpr68**

File: src/test/java/org/entando/kubernetes/service/EntandoBundleUtilitiesTest.java
Comment (w-caffiero-entando) [inline src/test/java/org/entando/kubernetes/service/EntandoBundleUtilitiesTest.java:30]: @firegloves assert should check for EntandoComponentManagerException, also see the code smell related to this part

**Review by w-caffiero-entando — CHANGES_REQUESTED** [review]

**Comment by entando-jx** [comment]
@firegloves: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 75a5830f380dfa26ee8b6b6996b1dccf033a0292 | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando-k8s/entando-component-manager/PR-68/16.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@firegloves: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 0ef2f14bcfd2ce1433a61f950c97869dfc7c2ffa | [link](https://raw.githubusercontent.com/entando-k8s/devops-results/logs/jenkins-x/logs/entando-k8s/entando-component-manager/PR-68/15.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-component-manager-ecmpr68**

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-component-manager-ecmpr68**

**Review by w-caffiero-entando — APPROVED** [review]

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando-k8s/entando-component-manager/pull/68#pullrequestreview-511764915" title="Approved">w-caffiero-entando</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **firegloves**

If they are not already assigned, you can assign the PR to them by writing `/assign @firegloves` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando-k8s%2Fentando-component-manager).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando-k8s/entando-component-manager/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/entando-k8s/entando-component-manager/pull/68#pullrequestreview-511764915" title="Approved">w-caffiero-entando</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando-k8s%2Fentando-component-manager).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando-k8s/entando-component-manager/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
